src/parser.rs

```diff
@@ -12,10 +12,10 @@ use crate::lexer::{Span, Token, TokenContents};
 use crate::typechecker::NumericConstant;
 
 macro_rules! trace {
-    ($x: expr) => {
+    ($($x:tt)*) => {
         #[cfg(feature = "trace")]
         {
-            println!("{}", $x)
+            println!($($x)*)
```

Comment:
The first argument to `println!()` should be a format string, like `clippy` is pointing out. Is there a reason to change this from what it was earlier? you could already do `trace!(format!(...))`

Reply:
This would allow trace to be written like println! and format! 

But I will undo it since it breaks the current traces. I didn't realize it at the time.